CMakeLists.txt

```diff
@@ -343,6 +343,9 @@ target_include_directories(clblast PUBLIC
                            $<INSTALL_INTERFACE:include>
                            ${API_INCLUDE_DIRS})
 
+# Tells CLBlast Includes that they are being used for internal usage
+target_compile_definitions(clblast PRIVATE CLBLAST_BUILDING_CLBLAST)
```

Comment:
Hmm, I find this define a bit weird. Can't you just make this PR without that new define, i.e. fix the CUDA API only? And then perhaps in a separate PR we can discuss how to make your linter happy - or - my preference - not at all: your linter isn't automatically applied anyway, so no need to introduce this verbosity and expose it to the user I would say?